go/stellar/global.go

```diff
@@ -163,6 +173,83 @@ func (s *Stellar) UpdateUnreadCount(ctx context.Context, accountID stellar1.Acco
 	return nil
 }
 
+type hasAcceptedDisclaimerDBEntry struct {
+	Version  int // 1
+	Accepted bool
+}
+
+// For a UV, accepted starts out false and transitions to true. It never becomes false again.
+// A cached true is returned, but a false always hits the server.
+func (s *Stellar) hasAcceptedDisclaimer(ctx context.Context) (bool, error) {
+	log := func(format string, args ...interface{}) {
+		s.G().Log.CDebugf(ctx, "Stellar.hasAcceptedDisclaimer "+format, args...)
+	}
+	uv, err := s.G().GetMeUV(ctx)
+	if err != nil {
+		return false, err
+	}
+	s.disclaimerLock.Lock()
+	defer s.disclaimerLock.Unlock()
+	// Check memory
+	memAccepted := s.disclaimerAccepted != nil && s.disclaimerAccepted.Eq(uv)
+	log("mem -> %v", memAccepted)
+	if memAccepted {
+		return true, nil
+	}
+	// Check disk
+	dbKey := libkb.DbKey{
+		Typ: libkb.DBStellarDisclaimer,
+		Key: uv.String(),
+	}
+	var dbEntry hasAcceptedDisclaimerDBEntry
+	found, err := s.G().LocalDb.GetInto(&dbEntry, dbKey)
+	log("disk -> [found:%v err:(%v) v:%v accepted:%v]", found, err, dbEntry.Version, dbEntry.Accepted)
+	if err == nil && found && dbEntry.Version == 1 && dbEntry.Accepted {
+		err = s.informAcceptedDisclaimerLocked(ctx)
+		if err != nil {
+			log("store -> err:(%v)", err)
+		}
+		return true, nil
+	}
+	// Check remote
+	accepted, err := remote.GetAcceptedDisclaimer(ctx, s.G())
+	log("remote -> [err:(%v) accepted:%v]", err, accepted)
+	if err != nil {
+		return false, err
+	}
+	if accepted {
+		err = s.informAcceptedDisclaimerLocked(ctx)
+		if err != nil {
+			log("store -> err:(%v)", err)
+		}
+	}
+	return accepted, nil
+}
+
+func (s *Stellar) informAcceptedDisclaimer(ctx context.Context) {
+	s.disclaimerLock.Lock()
+	defer s.disclaimerLock.Unlock()
+	_ = s.informAcceptedDisclaimerLocked(ctx)
+}
+
+func (s *Stellar) informAcceptedDisclaimerLocked(ctx context.Context) (err error) {
+	defer s.G().CTraceTimed(ctx, "Stellar.informAcceptedDisclaimer", func() error { return err })()
+	uv, err := s.G().GetMeUV(ctx)
```

Comment:
There's a very small chance uv could change in between your checks above and this line.  But it is there...

I'm fine with it as `keybase logout && keybase login someoneelse` would be very hard to squeeze into that time window, but you could pass uv in here...